File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py

def solution(cards1, cards2, goal):
    answer = "Yes"
    
    for word in goal:
        if len(cards1) >0 and word == cards1[0]: 
            cards1.pop(0)
        elif len(cards2) and word == cards2[0]:
            cards2.pop(0)
        else:
            answer="No"
            break
    
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "";
        
        Queue<String> q1 = new LinkedList<>();
        Queue<String> q2 = new LinkedList<>();
        
        
        for(int i = 0;i<cards1.length;i++){
            q1.add(cards1[i]);    
        }
        
        
        for(int i = 0;i<cards2.length;i++){
            q2.add(cards2[i]);    
        }
        
        for(int i = 0;i<goal.length;i++){
            String temp = goal[i];
            String top1 = q1.peek() == null ? "" : q1.peek();
            String top2 = q2.peek() == null ? "" : q2.peek();
            
            
            if(top1.equals(temp)){
                q1.remove();
            }else if(top2.equals(temp)){
                q2.remove();
            }else{
                return "No";
            }
        }
        
        return "Yes";
    }
}